fru_reader.c

```diff
@@ -4,28 +4,42 @@
 #include "fru_reader.h"
 #include "fatal.h"
 
-void safe_read(int fd, void *buffer, size_t length) {
+static void safe_read(int fd, void *buffer, size_t length) {
 	size_t total_bytes_read = 0;
 	while (total_bytes_read != length) {
 		ssize_t bytes_read = read(
 			fd, buffer + total_bytes_read, length - total_bytes_read);
 		if (bytes_read == -1)
 			fatal("Error reading file");
+		if (bytes_read == 0)
+			fatal("Reached end of file");
```

Comment:
I'd say here "File is too short" and maybe print the values too, for ease of debugging